src/pve_exporter/http.py

```diff
@@ -112,7 +112,24 @@ def __call__(self, request):
         return urls.dispatch(view_func, catch_http_exceptions=True)
 
 
-def start_http_server(config, port, address, collectors):
+class StandaloneGunicornApplication(gunicorn.app.base.BaseApplication):
+
+    def __init__(self, app, options=None):
+        self.options = options or {}
+        self.application = app
+        super().__init__()
+
+    def load_config(self):
+        config = {key: value for key, value in self.options.items()
+                  if key in self.cfg.settings and value is not None}
+        for key, value in config.items():
+            self.cfg.set(key.lower(), value)
```

Comment:
This is hardly readable. However, it follows the [upstream example](https://docs.gunicorn.org/en/latest/custom.html#custom-application) and I honestly have no better idea on how to translate the `options` dictionary into the gunicorn `cfg` format. Hence, I'm okay with it.

Reply:
I agree. At least I added a comment pointing out the source.